sdk/src/types/block/core.rs

```diff
@@ -95,7 +146,87 @@ impl BlockBuilder {
             return Err(Error::InvalidBlockLength(block_bytes.len()));
         }
 
-        Ok(block)
+        Ok((block, block_bytes))
+    }
+
+    /// Finishes the [`BlockBuilder`] into a [`Block`].
+    pub fn finish(self) -> Result<Block, Error> {
+        self._finish().map(|res| res.0)
+    }
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, From)]
+pub enum BlockType {
```

Comment:
I don't really like this name, it's breaking our consistency. I tried to figure out what we could change and saw that they called the wrapper `ProtocolBlock` in iota.go (https://github.com/iotaledger/iota.go/blob/serix-iota-core/block.go#L127). This may inform what it would generally be called so maybe a good idea to follow on this. Open to discussion.

Reply:
But I think this enum should be called `Block`

Reply:
I'm kind of inclined to move all the shared fields to the enum and just call it `Block`, which would fit with the other design like `Output`, which has variants that share fields but they each still have their own instance of it. How does that sound?

Reply:
I don't know if that would really work, because the difference with Output and all is that the type byte would be in the middle

Reply:
That's just a matter of the implementation of packable tho

Reply:
I'll just have to try it I think, I'm not sure